### PR TITLE
WEB-7741 Updating linting, testing and lodash

### PR DESCRIPTION
**Pull request by philmeehan**
#### What are the links to relevant tickets?
https://hxshortbreaks.atlassian.net/browse/WEB-7741

#### What does this PR do?
This is a PR to update this repo to the latest make-up and to upgrade lodash from the unsupported v2.

I have also swapped the tests to expect from should to match all our other repos and removed the spaces between brackets to match the coming make-up rules.

#### What functional/unit tests does this PR have?
Same as before

#### Is there anything a developer should be aware of when reviewing this?
This repo is about to enter maintenance mode as we have decided to move themeBlueprint over to handlebars but it may still need to be supported for theatreBlueprint so I thought it was worth this update.

Most of the diff will just be the spaces.

The only other functional difference is a change from _.include to _.includes and a couple of _.first to _.head (_.first is deprecated in v4 of lodash)

#### Screenshots (if appropriate)
---
#### Quality checklist (for PR author to complete BEFORE code review):
- [x] I've checked there is appropriate unit test coverage.
- [x] I've updated documentation with any changes to procedures.
- [x] I've checked this work against the requirements of the Jira.
- [x] This change passes all necessary tests (cross browser, automated or otherwise).
- [x] I am ready for this to be code reviewed, merged and tested.

#### Reviewer 1 @hxpaul
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [ ] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

#### Reviewer 2 @taijuten
- [x] I agree with the assumptions made in the quality checklist.
- [ ] I’ve witnessed the work behaving as expected.
- [x] I’ve checked for appropriate test coverage.
- [x] I’ve checked for coding anti-patterns.
- [x] I've checked this work against the requirements of the Jira.
- [x] I don’t believe this work introduces unnecessary technical debt.
- [x] +1 from me!

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,40 +1,48 @@
...
+  "main": "index",
+  "engines": {
+    "node": ">= 0.10.0"
+  },
+  "scripts": {
+    "ci": "istanbul cover _mocha -- -R xunit > $CIRCLE_TEST_REPORTS/unit/results.xml",
+    "postci": "istanbul check-coverage",
+    "lint": "make-up lib helpers",
+    "prerelease": "node_modules/deployment-helpers/nodeApps/preRelease.sh",
+    "start": "scripts/startup.sh",
```
Comment (hxpaul) [inline package.json]: there's no start script here could ditch this

File: helpers/filterSeats.js [hunk 2]
```diff
@@ -15,11 +14,11 @@ module.exports = function( dust ) {
   * @example {@_filterSeats packageRates=packageRatesReply.packageRates venueProducts=venueProductsReply.venueProducts roomRates=packageRatesReply.linked.roomRates ticketRates=packageRatesReply.linked.ticketRates} {/_filterSeats} output loop of sections with seats inside sorted by price
   */
 
-  dust.helpers._filterSeats = function( chunk, context, bodies, params ) {
+  dust.helpers._filterSeats = function(chunk, context, bodies, params) {
     params = params || {};
 
     var packageRates = params.packageRates;
-    var venueProduct = _.first( params.venueProducts );
+    var venueProduct = _.head(params.venueProducts);
```
Comment (hxpaul) [inline helpers/filterSeats.js]: worth a simple test that at least calls these that have actually changed, to check that we can call them without error with this new method and this version of lodash. Probably worth making sure all lodash lines are covered really when we know there are breaking changes. I will do a pr to this branch with an example test.

**Comment by hxpaul** [comment]
handy for this pr
```
alias giff='git diff --ignore-space-at-eol -b -w --ignore-blank-lines'
giff master
```

File: helpers/filterSeats.js (hunk 2 above)
Comment (hxpaul) [inline helpers/filterSeats.js]: If you think it's not worth adding this let me know, I'm not familiar with this repo and will go with your judgement, especially if we're going to archive this one...

File: package.json (hunk 1 above)
Comment (philmeehan) [inline package.json]: Updated

File: helpers/filterSeats.js (hunk 2 above)
Comment (philmeehan) [inline helpers/filterSeats.js]: I've had a think over the weekend on this. We are about to start the process to dump dust for handlebars. The test that you have suggested isn't really testing the intention of the code, and due to time constraints I won't be able to add a proper set of unit tests for these helpers.

I can see it is working by npm linking the-wall-tempater-dustjs and will make sure we properly QA the sites that the Wall serves.

When we need similar functionality in Handlebars we will make every helper is properly unit tested.

**Pull request merged**